### PR TITLE
Add TranscriptLine field to Comment

### PR DESCRIPTION
**Pull request by n-wach**
This is used for saving where to add indicators.

The discussion currently has no way of getting the current transcript line, so there is a TODO.  Once there is a way to get the current transcript line, the new comment post request can be modified to use it, and this feature should be fully working.

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java [hunk 1]
```diff
@@ -86,19 +89,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     final Entity commentEntity;
     long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
     Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
-    // TODO: Require non-zero content length. See: #183.
+    // TODO: Validate that the transcript line contains the timestampMs.
+    long transcriptId = Long.parseLong(request.getParameter(PARAM_TRANSCRIPT_LINE));
+    Key transcriptLine = KeyFactory.createKey(TranscriptLineUtil.KIND, transcriptId);
```
Comment (jaysaleh) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:94]: Can you name this `transcriptLineKey`?

Comment (jaysaleh) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:94]: Just a note, in some places we have 'key' in the name of the key. Like in `CommentUtil.java` this would be named `transcriptLineKey`. Can we stay consistent to avoid confusing in the naming, can be done in a seperate pr.

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:94]: Good point, I'll make the change in this one.  Thanks!

File: src/main/java/com/googleinterns/zoomtube/data/Comment.java [hunk 2]
```diff
@@ -49,6 +49,11 @@
    */
   public abstract Optional<Long> timestampMs();
 
+  /**
+   * Returns the key of the transcript line where an indicator should appear.
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/data/Comment.java:53]: So for encapsulation the Comment knows nothing about the TranscriptLine indicator.  I would rephrase this as:

Returns the transcript line key for which this comment is for.

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java [hunk 3]
```diff
@@ -85,20 +88,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     final Entity commentEntity;
     long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
-    Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
-    // TODO: Require non-zero content length. See: #183.
+    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, lectureId);
+    // TODO: Validate that the transcript line contains the timestampMs.
+    long transcriptId = Long.parseLong(request.getParameter(PARAM_TRANSCRIPT_LINE));
+    Key transcriptLineKey = KeyFactory.createKey(TranscriptLineUtil.KIND, transcriptId);
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:94]: If the lecture has no transcript, then does this still returns a transcriptId?

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java [hunk 4]
```diff
@@ -85,20 +88,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     final Entity commentEntity;
     long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE));
-    Key lecture = KeyFactory.createKey(LectureUtil.KIND, lectureId);
-    // TODO: Require non-zero content length. See: #183.
+    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, lectureId);
+    // TODO: Validate that the transcript line contains the timestampMs.
+    long transcriptId = Long.parseLong(request.getParameter(PARAM_TRANSCRIPT_LINE));
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:93]: I believe 'transcriptId' should be renamed 'transcriptLineId'. This is the id of the transcript line and not the transcript, right?

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java [hunk 5]
```diff
@@ -113,6 +119,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     if (request.getParameter(PARAM_TYPE) == null) {
       return Optional.of(ERROR_MISSING_COMMENT_TYPE);
     }
+    if (request.getParameter(PARAM_TRANSCRIPT_LINE) == null) {
+      return Optional.of(ERROR_MISSING_TRANSCRIPT_LINE);
+    }
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java:124]: Is this an error though? It is possible for a Lecture to have no transcript right?

File: src/main/webapp/view/discussion/discussion-manager.js [hunk 6]
```diff
@@ -88,22 +89,24 @@ export default class DiscussionManager {
...
     await this.postComment(content, {
       [DiscussionManager.#PARAM_TIMESTAMP]: timestampMs,
       [DiscussionManager.#PARAM_TYPE]: type,
+      [DiscussionManager.#PARAM_TRANSCRIPT_LINE]: transcriptLineId,
     });
   }
 
   /**
-   * Posts `content` as a reply to `parentId`.
+   * Posts `content` as a reply to `parentId` with the specified `transcriptId`.
```
Comment (billie-cat) [inline src/main/webapp/view/discussion/discussion-manager.js:103]: 'transcriptId' is a type.the field is transcriptLineId.

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java (hunk 4 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java]: ah yes, fixed!

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java (hunk 3 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java]: talked to amy about how we want to handle this, and had decided to make this param optional

File: src/main/java/com/googleinterns/zoomtube/data/Comment.java (hunk 2 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/data/Comment.java]: done!

File: src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java (hunk 5 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/DiscussionServlet.java]: talked to amy about how we want to handle this, and had decided to make this param optional

File: src/main/webapp/view/discussion/discussion-manager.js (hunk 6 above)
Comment (n-wach) [reply src/main/webapp/view/discussion/discussion-manager.js:103]: ah, whoops, fixed!

File: src/main/java/com/googleinterns/zoomtube/data/Comment.java [hunk 7]
```diff
@@ -75,6 +81,8 @@ public static Builder builder() {
     public abstract Builder setParentKey(Optional<Key> parentKey);
     public abstract Builder setTimestampMs(long timestampMs);
     public abstract Builder setTimestampMs(Optional<Long> timestampMs);
+    public abstract Builder setTranscriptLineKey(Key transcriptLineKey);
+    public abstract Builder setTranscriptLineKey(Optional<Key> transcriptLineKey);
```
Comment (billie-cat) [inline src/main/java/com/googleinterns/zoomtube/data/Comment.java:85]: Why are these methods duplicated? You only need 1.

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/data/Comment.java:85]: I am having some compiler errors when I only include one of them...

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/data/Comment.java:84]: Removed the non-optional one after talking offline
